docs/design/gateway/lwM2M-gateway.md

```diff
@@ -0,0 +1,211 @@
+# LwM2M Gateway design
+
+## Why need LwM2M Gateway
+
+A telemetry service typically pushes data from devices to a data server but does not have the capability to pull data from the device or post data from the cloud to the device. LwM2M (Lightweight Machine to Machine) protocol, however, usually requires both these features. To support this, an LwM2M Gateway is necessary to enable the deviceShifu to adapt to the LwM2M protocol, supporting pull data calls from the server and auto-pushing data to the server.
+
```

Comment:
supporting pull data calls from the server and auto-pushing data to the server. didn't contain the "post data from the cloud to the device" statement in the previous text.